### PR TITLE
Optional space

### PR DESCRIPTION
**Pull request by rishijain**
For this https://github.com/yabwe/medium-editor/issues/146

### CODE REVIEW

**Comment by j0k3r** [comment]
Thanks for your PR!
If I understand correctly, this _only_ fix "_Enter a second space between the words._", right?

**Comment by rishijain** [comment]
It will also block spaces at beginning and end of the element.
```javascript
if ((textContent[caretPositions.left - 1] === undefined) || (textContent[caretPositions.left - 1] === ' ') || (textContent[caretPositions.left] === undefined)) {
           event.preventDefault();
 }
```

For starting of the element, 
```javascript
textContent[caretPositions.left - 1] === undefined
```
this should catch this.

and similarly this should catch when cursor is at the end of the element.
```javascript
textContent[caretPositions.left] === undefined
```

**Comment by j0k3r** [comment]
Ok sorry I was checking too fast

**Comment by nmielnik** [comment]
@rishijain would you mind adding documentation for the new option & event?

The documentation for options is in both the [README.md](https://github.com/yabwe/medium-editor/blob/master/README.md), as well as the [OPTIONS.md](https://github.com/yabwe/medium-editor/blob/master/OPTIONS.md).

The documentation for events is in [CUSTOM-EVENTS.md](https://github.com/yabwe/medium-editor/blob/master/CUSTOM-EVENTS.md)

**Comment by rishijain** [comment]
@nmielnik  Done.

File: OPTIONS.md [hunk 1]
```diff
@@ -142,6 +144,15 @@ Enables/disables the use of the return-key. You can also set specific element be
 Allows/disallows two (or more) empty new lines. You can also set specific element behavior by using setting a data-disable-double-return attribute.
 
 ***
+#### `disableExtraSpaces`
+**Default:** `false`
+
+When set to true, it disallows spaces at the beginning and end of the element. Also it disallows entering 2 consecutive spaces between 2 words.
+
+
+***
```
Comment (j0k3r) [inline OPTIONS.md]: This extra line isn't needed, right?

**Comment by nmielnik** [comment]
Thanks @rishijain! Could you add your 'editableKeydownSpace` event into `CUSTOM-EVENTS.md` as well?

File: OPTIONS.md (hunk 1 above)
Comment (rishijain) [inline OPTIONS.md]: Sorry my bad. Removed.

**Comment by rishijain** [comment]
Sorry hurried into it earlier. Added now.

**Comment by nmielnik** [comment]
Looks awesome, thanks!

**Comment by alvarotrigo** [comment]
This doesn't seem to be solved in the latest version 

https://cdnjs.cloudflare.com/ajax/libs/medium-editor/5.23.3/js/medium-editor.min.js

**Comment by alvarotrigo** [comment]
Ah, nope. It does work, but I was using ` white-space: pre-wrap; ` for my element and that was messing up things.

**Pull request merged**